### PR TITLE
feat(client): introduce includeExtensionProperties subscription option

### PR DESCRIPTION
**Pull request by mboskamp**
Related to CAM-12081

### CODE REVIEW

**Comment by mboskamp** [comment]
The tests can not succeed until the corresponding changes to the REST API are merged.

File: client/src/it/java/org/camunda/bpm/client/task/ExternalTaskCustomExtensionPropertiesIT.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    // then
+    clientRule.waitForFetchAndLockUntil(() -> !handler.getHandledTasks().isEmpty());
+    ExternalTask task = handler.getHandledTasks().get(0);
+
+    assertThat(task.getExtensionProperties()).isEmpty();
+  }
+
+  @Test
+  public void shouldNotReceiveCustomExtensionProperties() {
```
Comment (tmetzke) [inline client/src/it/java/org/camunda/bpm/client/task/ExternalTaskCustomExtensionPropertiesIT.java:100]: That needs to be changed to reflect the "never null" contract for the properties map.
Might be that you need to change something in the implementation as well to fulfill that contract.

File: client/src/main/java/org/camunda/bpm/client/task/ExternalTask.java [hunk 2]
```diff
@@ -195,6 +196,30 @@
...
+   * fetched (e.g. by calling
+   * {@link TopicSubscriptionBuilder#includeExtensionProperties(boolean)}) to be
+   * available.
+   * 
+   * If no extension properties are available the returned map will be empty.
+   * 
+   * @return a map of available extension properties which will be empty if no
+   *         property is available or extension properties were not fetched
+   *         (e.g. by calling
+   *         {@link TopicSubscriptionBuilder#includeExtensionProperties(boolean)})
```
Comment (tmetzke) [inline client/src/main/java/org/camunda/bpm/client/task/ExternalTask.java:220]: I am not that big a fan of duplicated information, the longer explanation says it all.
The "never null" part would be more interesting I guess.
```suggestion
   * @return a map of available extension properties, never <code>null</code>
```

**Comment by mboskamp** [comment]
The tests will fail again, because the engine artifact builds are not done yet. I will re-trigger the tests once the artifacts are ready.

**Pull request merged**